Hazel/src/Platform/Windows/WindowsWindow.cpp

```diff
@@ -172,5 +222,50 @@ namespace Hazel {
 	{
 		return m_Data.VSync;
 	}
+	
+	void WindowsWindow::SetWindowMode(const WindowMode& mode)
+	{
+		if (mode == m_Data.Mode) return;
+		HZ_CORE_TRACE("Swapping to mode '{0}'", mode == WindowMode::Windowed ? "Windowed" : mode == WindowMode::Fullscreen ? "Fullscreen" : "Borderless");
+
+		GLFWmonitor* windowRenderScreen = glfwGetPrimaryMonitor(); // where the window will be rendered
+		const GLFWvidmode* baseVideoMode = glfwGetVideoMode(windowRenderScreen);
+
+		if (m_Data.Mode == WindowMode::Borderless)
+			glfwRestoreWindow(m_Window);
+
+		// Set the actual width and height
+		int width, height;
+		switch (mode)
+		{
+			case WindowMode::Fullscreen:
+			case WindowMode::Borderless:
+				width = baseVideoMode->width;
+				height = baseVideoMode->height;
+				break;
+			case WindowMode::Windowed:
+				width = m_Data.WindowedWidth;
+				height = m_Data.WindowedHeight;
+				break;
+		}
+
+		HZ_CORE_ASSERT(m_Window, "Failed to retrieve window.");
+		m_Data.Mode = mode;
+		glfwSetWindowMonitor(m_Window,
+		                     m_Data.Mode == WindowMode::Fullscreen ? windowRenderScreen : nullptr,
+		                     m_Data.Mode == WindowMode::Windowed ? m_Data.WindowedPos.x : 0,
+		                     m_Data.Mode == WindowMode::Windowed ? m_Data.WindowedPos.y : 0,
+		                     width, height,
+		                     baseVideoMode->refreshRate);
+		glfwSetWindowAttrib(m_Window, GLFW_DECORATED, m_Data.Mode == WindowMode::Windowed ? GLFW_TRUE : GLFW_FALSE);
+		glfwSetWindowAttrib(m_Window, GLFW_RESIZABLE, m_Data.Mode == WindowMode::Windowed ? GLFW_TRUE : GLFW_FALSE);
+		glfwSetWindowAttrib(m_Window, GLFW_FLOATING, m_Data.Mode == WindowMode::Fullscreen ? GLFW_TRUE : GLFW_FALSE);
+		glfwSetWindowAttrib(m_Window, GLFW_AUTO_ICONIFY, m_Data.Mode == WindowMode::Fullscreen ? GLFW_TRUE : GLFW_FALSE);
```

Comment:
We are getting extra sizing events when we change these attributes. It would be beter to call these attributes before the `glfwSetWindowMonitor` and surpress the sizing callback while setting these attributes. Before setting the Window we re-enable the sizing callback and get 1 (final) size callback for the new window.

Reply:
Fixed in https://github.com/TheCherno/Hazel/pull/108/commits/70ce9c5acb23dadca64684fdaee6970699b3970a